### PR TITLE
166468649 Fix Validation Issues

### PR DESCRIPTION
**Pull request by Ukhu**
#### What does this PR do?
Implement the validation changes raised during team pre-demo call and after the demo meeting

#### Description of Task to be completed?
  - Modify User validations to allow compound words for firstname and lastname
  - Modify User validations to allow up to 2 words for firstname and lastname
  - Modify Author validations to allow up to 3 words for authorName
  - Modify Author and User validations to allow names with apostrophe
  - Modify validations to disallow password with spaces
  - Remove capitaliseFirstLetter custom sanitation
  - Transform firstname and lastname input to lowercase
  - Fix checkForErrors middleware to respond with success or failure status instead of status code

#### How should this be manually tested?
  - pull this branch
  - run `npm test`
  - open Postman and test the validations on the signup, login and author route

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166468649](https://www.pivotaltracker.com/story/show/166468649)

#### Screenshots (if appropriate)
Screenshot (160)

#### Questions:
N/A

### CODE REVIEW

**Review by KvNGCzA — CHANGES_REQUESTED** [review]
Write tests to cover the new cases you just added.

**Review by Lundii — COMMENTED** [review]
Can you add the new validations you added in the PR so we know exactly what we are testing for.

**Comment by Ukhu** [comment]
> Write tests to cover the new cases you just added.

Done

**Comment by Ukhu** [comment]
> Can you add the new validations you added in the PR so we know exactly what we are testing for.

Alright. Done that

**Review by Lundii — COMMENTED** [review]
Working fine. Please rebase so I approve

**Comment by Ukhu** [comment]
> Working fine. Please rebase so I approve

Done

File: server/middlewares/UserValidator.js [hunk 1]
```diff
@@ -45,12 +45,12 @@ export default class UserValidator {
...
   static checkName(name) {
     return UserValidator.genericCheck(`${name}`)
-      .isAlpha()
-      .withMessage(`${name} can only contain alphabets`)
+      .trim()
       .isLength({ min: 2, max: 20 })
       .withMessage(`${name} must be at least 2 characters, and maximum 20`)
-      .blacklist(' ')
-      .customSanitizer(value => capitaliseFirstLetter(value));
+      .matches(/^[A-Za-z]+([ ]?|[-]?)[A-Za-z]+$/, 'g')
```
Comment (allebd) [inline server/middlewares/UserValidator.js]: What of for names like Shaquille **O'Neal** can that be valid

**Review by Lundii — DISMISSED** [review]

**Review by allebd — DISMISSED** [review]
Nice Work

**Review by henryade — DISMISSED** [review]

**Review by KvNGCzA — APPROVED** [review]

**Pull request merged**